mlir/include/Gradient/IR/GradientOps.td

```diff
@@ -105,10 +140,15 @@ def BackpropOp : Gradient_Op<"backprop", [AttrSizedOperandSegments,
             RankedTensorOf<[AnyFloat]>,
             MemRefOf<[AnyFloat]>
         ]>>:$cotangents,
-        OptionalAttr<AnyIntElementsAttr>:$diffArgIndices
+        OptionalAttr<AnyIntElementsAttr>:$diffArgIndices,
+        OptionalAttr<Builtin_StringAttr>:$requester
```

Comment:
I think rather than specifying who invoked it, which requires one additional logic hop to determine what we actually want, let's go straight to the desired property `results_needed`, `discard_results`, etc.

Reply:
That makes sense, I'll update it

Reply:
done